src/common/secured-property.ts

```diff
@@ -19,7 +19,7 @@ export type UnwrapSecured<T> = T extends Secured<infer P> ? P : T;
 export const isSecured = <T>(value: T | Secured<T>): value is Secured<T> =>
   isObject(value);
 
-export const unwrapSecured = <T>(value: T | Secured<T>): T =>
+export const unwrapSecured = <T>(value: T | Secured<T>): T | undefined =>
```

Comment:
Got it, it looks like the problem here is line 11 - 15:
```
export interface Secured<T> {
  readonly value?: T;
  readonly canRead: boolean;
  readonly canEdit: boolean;
}
```

`readonly value?: T` is implying that `value` can be undefined. How should we approach that? I can set `value` to not be undefinable but it will probably cause some downstream effects in the property-updater.service. Lemme know your thoughts

Reply:
Hmm ok yeah I see. Ok I'm fine leaving it as you've changed it. I think it does make the most sense in the context of the codebase. 👍

Reply:
Sounds good!